oauthproxy.go

```diff
@@ -955,23 +956,64 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 
 // authOnlyAuthorize handles special authorization logic that is only done
 // on the AuthOnly endpoint for use with Nginx subrequest architectures.
-//
-// TODO (@NickMeves): This method is a placeholder to be extended but currently
-// fails the linter. Remove the nolint when functionality expands.
-//
-//nolint:gosimple
 func authOnlyAuthorize(req *http.Request, s *sessionsapi.SessionState) bool {
-	// Allow secondary group restrictions based on the `allowed_groups`
-	// querystring parameter
-	if !checkAllowedGroups(req, s) {
-		return false
+	constraints := []func(*http.Request, *sessionsapi.SessionState) bool{
+		checkAllowedGroups,
+		checkAllowedEmailDomains,
+	}
+
+	for _, constraint := range constraints {
+		if !constraint(req, s) {
+			return false
+		}
 	}
 
 	return true
 }
 
+func extractAllowedEntities(req *http.Request, field string) map[string]struct{} {
```

Comment:
I was a bit confused about what we were trying to achieve here, can we add a comment that explains what the inputs and outputs are at a minimum

Reply:
I added a comment based on @NickMeves recommendations.
See https://github.com/oauth2-proxy/oauth2-proxy/pull/1286#discussion_r702448582